lib/user/sfBasicSecurityUser.class.php

```diff
@@ -251,12 +251,6 @@ public function initialize(sfEventDispatcher $dispatcher, sfStorage $storage, $o
       $this->options['timeout'] = 1800;
     }
```

Comment:
This logic need also to be copied before access of `$this->options['timeout']` on `sfSessionStorage`.

-> https://github.com/FriendsOfSymfony1/symfony1/blob/0f6bdd8814e33a3713fc36df780dfe583d8033b1/lib/storage/sfSessionStorage.class.php#L56-L66

Reply:
thanks @alquerci  for your review :)

> Need to check deeper but on a first approximation there can be a BC break regarding session timeout.

I wonder how's that? Current code on `sfBasicSecurityUser` is clearly no-op. It just don't send a warning, but it's not working for sure.

> Also need to propagate this new factory parameter to all other session classes

no needs, as all session classes derives from `sfSessionStorage` and calls to `parent::initialize(...)`

>where both configuration needs to be in synchronization

A `user` can get the option from a `storage` and if it exists use it (check updated PR).

Regarding the name of parameter, I'd rather leave it as `timeout` as it's more meaningful when we are reading the config.

Reply:
> A user can get the option from a storage and if it exists use it

It is exactly the inverse to do as the current timeout is given on user scope.

This behaviour is not going to change.

---

### Actions

1. Keep the same configuration schema.
2. To be able to fetch the timeout configuration from `user` factory configuration.
3. Before `session_start()` if `timeout` is greater than `gc_maxlifetime` then `gc_maxlifetime = timeout`

### Issues to solve

* `user` depends on `storage` but `storage` depends on `user` config
  - Track to patch -> on the `sfFactoryConfigHandler`

Reply:
you killing me, as always :stuck_out_tongue: alright let me check.

Reply:
btw, @alquerci, can you advise me how to run tests on the library properly? :pray:

Reply:
@spirit-q2 Look at how tests are executed on travis configuration.

Using official docker php:7.2-cli image with following extensions

* pdo
* pdo_mysql                                                                                                                                              
* mbstring

And link it to a MySql database with an empty database named `test`.
The official mysql:5.5 docker image can make the stuff with env `MYSQL_ALLOW_EMPTY_PASSWORD=yes`.